Geometry/RPCGeometryBuilder/src/RPCGeometryParsFromDD.cc

```diff
@@ -75,18 +102,84 @@ void RPCGeometryParsFromDD::buildGeometry(DDFilteredView& fview,
     rota.GetComponents(x, y, z);
     std::vector<double> pars;
     if (dpar.size() == 3) {
-      const double width = dpar[0];
-      const double length = dpar[1];
-      const double thickness = dpar[2];
+      const double width = geant_units::operators::convertMmToCm(dpar[0]);
+      const double length = geant_units::operators::convertMmToCm(dpar[1]);
+      const double thickness = geant_units::operators::convertMmToCm(dpar[2]);
+      edm::LogVerbatim("RPCGeometryParsFromDD")
+          << " (2) dpar.size() == 3, width: " << width << " length: " << length << " thickness: " << thickness;
       pars = {width, length, thickness, numbOfStrips.doubles()[0]};
     } else {
-      pars = {
-          dpar[4] /*b/2*/, dpar[8] /*B/2*/, dpar[0] /*h/2*/, 0.4, numbOfStrips.doubles()[0] /*h/2*/
-      };
+      const double dpar4 = geant_units::operators::convertMmToCm(dpar[4]);
+      const double dpar8 = geant_units::operators::convertMmToCm(dpar[8]);
+      const double dpar0 = geant_units::operators::convertMmToCm(dpar[0]);
+      pars = {dpar4, dpar8, dpar0, 0.4, numbOfStrips.doubles()[0]};
+      edm::LogVerbatim("RPCGeometryParsFromDD")
+          << " (3), else, dpar[4]: " << dpar4 << " dpar[8]: " << dpar8 << " dpar[0]: " << dpar0;
     }
-
-    const std::vector<double> vtra = {tran.x(), tran.y(), tran.z()};
+    const std::vector<double> vtra = {geant_units::operators::convertMmToCm(tran.x()),
```

Comment:
This change is worrisome. The old DD code should not need to be changed. Why would the units convention change now for old DD?

Reply:
@cvuosalo : the same code is present in the RPCGeometryBuilder Class. Without it in the Log file I found:

RPCGeometryParsFromDD, tran.x(): 4136.75 tran.y(): 376 tran.z(): -4704.5 // for DD
RPCGeometryParsFromDD, tran.x(): 413.675 tran.y(): 37.6 tran.z(): -470.45 // for DD4Hep

so I think that's geant4_units are necessary